src/gtfs/write.rs

```diff
@@ -55,13 +55,26 @@ pub fn write_transfers(path: &path::Path, transfers: &Collection<NtfsTransfer>)
 pub fn write_agencies(
     path: &path::Path,
     networks: &CollectionWithId<objects::Network>,
+    ticketing_deep_links: &TicketingDeepLinks,
 ) -> Result<()> {
     info!("Writing agency.txt");
     let path = path.join("agency.txt");
     let mut wtr =
         csv::Writer::from_path(&path).with_context(|| format!("Error reading {:?}", path))?;
-    for n in networks.values() {
-        wtr.serialize(Agency::from(n))
+    for network in networks.values() {
+        let mut agency = Agency::from(network);
+        if !ticketing_deep_links.is_empty() {
+            agency.ticketing_deep_link_id = network
+                .fare_url
+                .as_ref()
+                .and_then(|fare_url| {
+                    ticketing_deep_links
+                        .get(fare_url)
+                        .map(|ticketing_deep_link| ticketing_deep_link.id.clone())
+                })
+                .or_else(|| Some(String::new()));
```

Comment:
Do you need to have an empty String here ?
If I understand `agency.ticketing_deep_link_id` may have 3 outputs

1. `Some("a_string_not_empty")` if `ticketing_deep_links IS NOT empty` and `ticketing_deep_links.get(fare_url)` exists
2. `Some("")` if `ticketing_deep_links IS NOT empty` and `ticketing_deep_links.get(fare_url)` DOES NOT exist
2. `None` if `ticketing_deep_links IS empty`

I think if you have something like

```rust
agency.ticketing_deep_link_id = network
    .fare_url
    .as_ref()
    .and_then(|fare_url| {
        ticketing_deep_links
        .get(fare_url)
        .map(|ticketing_deep_link| ticketing_deep_link.id.clone())
    });
}
```
It will return `Some("something")` or `None` and `None` will be serialized to `""`